Zend/zend_language_scanner.l

```diff
@@ -1563,6 +1658,15 @@ NEWLINE ("\r"|"\n"|"\r\n")
 
 
 <ST_IN_SCRIPTING>"}" {
+	if(in_class) {
+		if(--in_class == 1) {
+			in_class = 0;	
+		}
+		if(in_use_list) {
+			in_use_list = 0;
+			in_trait_conflict = 0;
+		}
+	}
```

Comment:
This only fails if "${$a}" (string interpolation) is part of `ST_IN_SCRIPTING` state. I can't check now as I'm far from working computer. 

But if your question is just rhetorical and you already know the answer is yes, the solution would be to conditionally `in_class++` [here](https://github.com/php/php-src/pull/1054/files#diff-93ad74868f98ff7232ebea00007c8b7fR1654) too  + some other things I can't envision right now.